src/aws_sdk_dynamodbstore.rs

```diff
@@ -257,101 +263,167 @@ impl Backend {
 
 #[async_trait]
 impl super::Backend for Backend {
+    #[tracing::instrument(skip_all, fields(key, consistent = !self.allow_eventually_consistent_reads, consumed_rcu, otel.status_code, error.msg, otel.span_kind = "client"))]
     async fn get<'a, K: Key<'a>>(&self, key: K) -> Result<Option<Value>> {
+        let key = key.into();
+        let span = tracing::Span::current();
+        span.record("key", tracing::field::debug(&key));
+
         let result = self
             .client
             .get_item()
             .consistent_read(!self.allow_eventually_consistent_reads)
-            .set_key(Some(composite_key(&key.into(), NO_SORT_KEY)))
+            .set_key(Some(composite_key(&key, NO_SORT_KEY)))
             .table_name(self.table_name.clone())
             .send()
-            .await?;
+            .await
+            .spanify_err()?;
+
+        record_rcu(&result.consumed_capacity, &span);
+
         Ok(result.item.and_then(|mut item| item.remove("v")).and_then(|v| match v {
             AttributeValue::B(b) => Some(b.into_inner().into()),
             _ => None,
         }))
     }
 
+    #[tracing::instrument(skip_all, fields(key, consumed_wcu, otel.status_code, error.msg, otel.span_kind = "client"))]
     async fn set<'a, 'b, K: Key<'a>, V: Into<Arg<'b>> + Send>(&self, key: K, value: V) -> Result<()> {
-        self.client
+        let key = key.into();
+        let span = tracing::Span::current();
+        span.record("key", tracing::field::debug(&key));
+
+        let result = self
+            .client
             .put_item()
             .table_name(self.table_name.clone())
-            .set_item(Some(new_item(&key.into(), NO_SORT_KEY, vec![("v", attribute_value(value))])))
+            .set_item(Some(new_item(&key, NO_SORT_KEY, vec![("v", attribute_value(value))])))
             .send()
-            .await?;
+            .await
+            .spanify_err()?;
+
+        record_wcu(&result.consumed_capacity, &span);
+
         Ok(())
     }
 
+    #[tracing::instrument(skip_all, fields(key, consumed_wcu, otel.status_code, error.msg, otel.span_kind = "client"))]
     async fn set_eq<'a, 'b, 'c, K: Key<'a>, V: Into<Arg<'b>> + Send, OV: Into<Arg<'c>> + Send>(&self, key: K, value: V, old_value: OV) -> Result<bool> {
+        let key = key.into();
+        let span = tracing::Span::current();
+        span.record("key", tracing::field::debug(&key));
+
         match self
             .client
             .put_item()
             .table_name(self.table_name.clone())
-            .set_item(Some(new_item(&key.into(), NO_SORT_KEY, vec![("v", attribute_value(value))])))
+            .set_item(Some(new_item(&key, NO_SORT_KEY, vec![("v", attribute_value(value))])))
             .condition_expression("v = :v")
             .expression_attribute_values(":v", attribute_value(old_value))
             .send()
```

Comment:
I believe this and other sites that call `record_wcu` or `record_rcu` need `returned_consumed_capacity` to be effective. (It'd be nice to have a test for its effect, if you can find a way to add one. Maybe by setting up a custom tracing subscriber?)